File: server/src/main/java/au/com/codeka/warworlds/server/store/base/StoreReader.java
package au.com.codeka.warworlds.server.store.base;

import au.com.codeka.warworlds.server.store.StoreException;
import java.sql.SQLException;
import javax.annotation.Nullable;
import org.sqlite.javax.SQLiteConnectionPoolDataSource;

/** A helper class for reading from the data store. */
public class StoreReader extends StatementBuilder<StoreReader> {
  StoreReader(SQLiteConnectionPoolDataSource dataSource, @Nullable Transaction transaction) {
    super(dataSource, transaction);
  }

  public QueryResult query() throws StoreException {
    execute();
    try {
      return new QueryResult(transaction == null ? conn : null, stmt.getResultSet());
    } catch (SQLException e) {
      throw new StoreException(e);
    }
  }

  @Override
  public void close() {
    // The QueryResult should close us, nothing to do here.
  }
}

File: server/src/main/java/au/com/codeka/warworlds/server/store/base/StoreReader.kt
package au.com.codeka.warworlds.server.store.base

import au.com.codeka.warworlds.server.store.StoreException
import org.sqlite.javax.SQLiteConnectionPoolDataSource
import java.sql.SQLException

/** A helper class for reading from the data store.  */
class StoreReader
    internal constructor(dataSource: SQLiteConnectionPoolDataSource, transaction: Transaction?)
    : StatementBuilder<StoreReader>(dataSource, transaction) {

  fun query(): QueryResult {
    execute()
    return try {
      QueryResult(if (transaction == null) conn else null, stmt.resultSet)
    } catch (e: SQLException) {
      throw StoreException(e)
    }
  }

  override fun close() {
    // The QueryResult should close us, nothing to do here.
  }
}
